napari/layers/image/_image_base_mouse_bindings.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+
+from ...utils.geometry import point_in_bounding_box
+
+
+def on_embedded_plane_drag(layer, event):
+    """Shift a rendered plane along its normal vector.
+
+    This function will shift a plane along its normal vector when the plane is
+    clicked and dragged. The general strategy is to
+    1) find both the plane normal vector and the mouse drag vector in canvas
+    coordinates
+    2) calculate how far to move the plane in canvas coordinates, this is done
+    by projecting the mouse drag vector onto the (normalised) plane normal
+    vector
+    3) transform this drag distance (canvas coordinates) into data coordinates
+    4) update the plane position
+    """
+    # local import to avoid circular import
+    from ...viewer import current_viewer
```

Comment:
eek! I don't love this, we've now leaked the whole viewer into our layers which we've been trying to keep from knowing about the viewer. i will look bellow at why this was needed and how we can avoid

Reply:
I agree this needs to be fixed - I hadn't considered the potential problems discussed in #3147